server/src/main/java/org/elasticsearch/action/admin/cluster/desirednodes/UpdateDesiredNodesRequest.java

```diff
@@ -42,18 +43,24 @@ public class UpdateDesiredNodesRequest extends AcknowledgedRequest<UpdateDesired
     }
 
     public UpdateDesiredNodesRequest(String historyID, long version, List<DesiredNode> nodes) {
```

Comment:
I would prefer to remove this constructor, that way we won't miss any necessary changes in tests, i.e. there's some coverage missing in `UpdateDesiredNodesRequestSerializationTests#mutateInstance`